source/opt/dead_branch_elim_pass.cpp

```diff
@@ -440,28 +441,34 @@ Pass::Status DeadBranchElimPass::Process() {
 }
 
 Instruction* DeadBranchElimPass::FindFirstExitFromSelectionMerge(
-    uint32_t start_block_id, uint32_t merge_block_id, uint32_t loop_merge_id) {
+    uint32_t start_block_id, uint32_t merge_block_id, uint32_t loop_merge_id,
+    uint32_t loop_continue_id) {
   // To find the "first" exit, we follow branches looking for a conditional
   // branch that is not in a nested construct and is not the header of a new
   // construct.  We follow the control flow from |start_block_id| to find the
   // first one.
-  while (start_block_id != merge_block_id) {
+  while (start_block_id != merge_block_id && start_block_id != loop_merge_id &&
+         start_block_id != loop_continue_id) {
     BasicBlock* start_block = context()->get_instr_block(start_block_id);
     Instruction* branch = start_block->terminator();
     uint32_t next_block_id = 0;
     switch (branch->opcode()) {
       case SpvOpBranchConditional:
         next_block_id = start_block->MergeBlockIdIfAny();
         if (next_block_id == 0) {
-          // If a possible target is the |loop_merge_id|, which is not the
-          // current merge node, then we have to continue the search with the
-          // other target.
+          // If a possible target is the |loop_merge_id| or |loop_continue_id|,
+          // which are not the current merge node, then we continue the search
+          // with the other target.
           for (uint32_t i = 1; i < 3; i++) {
             if (branch->GetSingleWordInOperand(i) == loop_merge_id &&
                 loop_merge_id != merge_block_id) {
               next_block_id = branch->GetSingleWordInOperand(3 - i);
               break;
             }
+            if (branch->GetSingleWordInOperand(i) == loop_continue_id) {
```

Comment:
The comment doesn't match this code. The loop continue block could be the same block as the current merge block.

Reply:
Fixed.  However, I don't think it makes a difference either way.